src/main/kotlin/nl/info/zac/productaanvraag/ProductaanvraagService.kt

```diff
@@ -230,6 +240,39 @@ class ProductaanvraagService @Inject constructor(
         return initiatorBetrokkene
     }
 
+    private fun Betrokkene.shouldBeSkipped(
```

Comment:
Minor thing but I had to look twice at what the logic was here: if it should be skipped, return a string, otherwise return null. Not immediately clear. I would personally just return a Boolean, because that is what the function name suggests anyway and just log the warnings inside this function. Then you could make the `forEach` loop above a bit cleaner by first filtering on `it.shouldBeSkipped(..)` and only then doing the `forEach`. The way it is now with breaking out of the forEach loop does not feel very nice.

Reply:
Thanks @edgarvonk! Applied your proposal. Looks better